### PR TITLE
fix: deeptools bamcoverage fix

### PR DESCRIPTION
**Pull request by jeffhsu3**
Fixes a small issue when including the genome and read_length pulling from the pre-computed effective genome sizes.  The returned genome size is an int rather than a string.  

### QC

* [x] I confirm that:

For all wrappers added by this PR, 

* there is a test case which covers any introduced changes,
* `input:` and `output:` file paths in the resulting rule can be changed arbitrarily,
* either the wrapper can only use a single core, or the example rule contains a `threads: x` statement with `x` being a reasonable default,
* rule names in the test case are in [snake_case](https://en.wikipedia.org/wiki/Snake_case) and somehow tell what the rule is about or match the tools purpose or name (e.g., `map_reads` for a step that maps reads),
* all `environment.yaml` specifications follow [the respective best practices](https://stackoverflow.com/a/64594513/2352071),
* the `environment.yaml` pinning has been updated by running `snakedeploy pin-conda-envs environment.yaml` on a linux machine,
* wherever possible, command line arguments are inferred and set automatically (e.g. based on file extensions in `input:` or `output:`),
* all fields of the example rules in the `Snakefile`s and their entries are explained via comments (`input:`/`output:`/`params:` etc.),
* `stderr` and/or `stdout` are logged correctly (`log:`), depending on the wrapped tool,
* temporary files are either written to a unique hidden folder in the working directory, or (better) stored where the Python function `tempfile.gettempdir()` points to (see [here](https://docs.python.org/3/library/tempfile.html#tempfile.gettempdir); this also means that using any Python `tempfile` default behavior works),
* the `meta.yaml` contains a link to the documentation of the respective tool or command,
* `Snakefile`s pass the linting (`snakemake --lint`),
* `Snakefile`s are formatted with [snakefmt](https://github.com/snakemake/snakefmt),
* Python wrapper scripts are formatted with [black](https://black.readthedocs.io).
* Conda environments use a minimal amount of channels, in recommended ordering. E.g. for bioconda, use (conda-forge, bioconda, nodefaults, as conda-forge should have highest priority and defaults channels are usually not needed because most packages are in conda-forge nowadays).

### CODE REVIEW

**Comment by fgvieira** [comment]
Can you add the new test to `test.py`?

File: bio/deeptools/bamcoverage/wrapper.py [hunk 1]
```diff
@@ -74,7 +74,9 @@
     read_length = snakemake.params.get("read_length")
     if genome and read_length:
         effective_genome_size = "--effectiveGenomeSize "
-        effective_genome_size += default_effective_genome_size[genome][str(read_length)]
+        effective_genome_size += str(
+            default_effective_genome_size[genome][str(read_length)]
+        )
```
Comment (fgvieira) [inline bio/deeptools/bamcoverage/wrapper.py:79]:
```
effective_genome_size = "--effectiveGenomeSize {}".format(default_effective_genome_size[genome][str(read_length)])
```

Comment (jeffhsu3) [reply bio/deeptools/bamcoverage/wrapper.py]: To handle cases where `genome` or `read_length` are not found in default_effective_genome_size, I've wrapped the line in a try-except block with a message on the available `genome` and `read_length` table from deeptools.  Also updated the values from the most recent deeptools and made the suggested changes to have it be more readable.

File: bio/deeptools/bamcoverage/wrapper.py [hunk 2]
```diff
@@ -73,10 +105,17 @@
...
+        except KeyError:
+            raise KeyError(
+                (
+                    "A valid genome and read_length need to be provided. See "
+                    "https://deeptools.readthedocs.io/en/latest/content/feature/effectiveGenomeSize.html"
+                )
+            )
 else:
-    effective_genome_size = "--effectiveGenomeSize " + str(effective_genome_size)
+    effective_genome_size = f"--effectiveGenomeSize {str(effective_genome_size)}"
```
Comment (fgvieira) [inline bio/deeptools/bamcoverage/wrapper.py:118]:
```suggestion
    effective_genome_size = f"--effectiveGenomeSize {effective_genome_size}"
```

**Review by fgvieira — APPROVED** [review]

**Pull request merged**